### DISCUSSION TITLE
Unexpected interval column can cause error

### DISCUSSION THREAD

**Discussion by racheleclapham** [humanpred/pknca#258] [Q&A] [answered]
Hi. I am really enjoying using the package! 

I am running into a problem that I can see was fixed in September 2023 (#238/239). I am trying to run pk.nca on my merged concentration and dose data.
 ie: my.data.manual<-PKNCAdata(conc_obj,dose_obj,intervals = my.intervals)
my.results.manual<-pk.nca(my.data.manual)

However, I am still receiving the following error:  
Caused by error in `vapply()`:
#> ! values must be length 1,
#>  but FUN(X[[2]]) result is length 0

I was wondering if you know how to fix this/what I am doing wrong? 

Many thanks
Rachel

**Comment by billdenney** [comment]
Hi Rachel,

I'm glad that you are enjoying PKNCA.

I can't reproduce the issue on my end.  Can you please send the full code and data you are using?  If the data are proprietary, can you please share the interval definition (`my.intervals`)?

**Comment by racheleclapham** [comment]
Hi Bill, 

Thank you for replying so quickly! 

This is the code I am using: 

```
d_conc<-ptall

d_dose <- d_conc[d_conc$time == 0,]
d_dose$time <- 0

conc_obj <- PKNCAconc(d_conc,
                      +dv~time|group+ID)
dose_obj <- PKNCAdose(d_dose,
                      amt~time|group+ID)

my.intervals<-data.frame(group=c("A","B","C","D","E","F","G","H","I","J"),start=0,end=Inf,cmax=T,ctrough=T, tmax=T, half.life=T,aucinf=T,auclast=T)
my.data.manual<-PKNCAdata(conc_obj,dose_obj,intervals = my.intervals)
my.results.manual<-pk.nca(my.data.manual)
knitr::kable(summary(my.results.manual))
```

I have also attached the data I am using. 

Thank you so much for your help
Rachel

**Comment by billdenney** [comment, marked as answer]
Hi Rachel,

I have not yet generated a new version for CRAN, so the September fix requires installing from GitHub (`remotes::install_github("billdenney/pknca")`).

Your issue is identical to the one in #238, the intervals use a column named `aucinf` rather than `aucinf.obs` (or similar).  This will work with the current version of PKNCA on CRAN:

```
library(PKNCA)

d_conc<-read.csv("https://github.com/billdenney/pknca/files/13478420/ptall.csv")

d_dose <- d_conc[d_conc$time == 0,]

conc_obj <- PKNCAconc(d_conc,
                      dv~time|group+ID)
dose_obj <- PKNCAdose(d_dose,
                      amt~time|group+ID)

my.intervals<-data.frame(start=0,end=Inf,cmax=T,ctrough=T, tmax=T, half.life=T,aucinf.obs=T,auclast=T)
my.data.manual<-PKNCAdata(conc_obj,dose_obj,intervals = my.intervals)
my.results.manual<-pk.nca(my.data.manual)
knitr::kable(summary(my.results.manual))
```

A couple of notes to help with future analyses:

* There was an extra "+" before "dv" in the PKNCAconc formula.  It doesn't hurt anything, but it isn't required.  (I made that change in the above code.)
* If all of your groups need the same NCA parameters calculated, then you do not need the group definitions in the intervals.  (I made that change in the above code.)

**Reply by racheleclapham** [reply]
Thank you so much, I really appreciate your help!
